### PR TITLE
RUMM-1429 Add nightly tests coverage visibility into the CI workflow

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

In this PR we are addressing several issues before finally reaching to the main goal - having a tests coverage tool for our nightly - tests to be executed in our CI workflow:

1. We addressed several corner cases in the test coverage computation script mostly related with the `apiSurface` file format.
2. We introduced the `NotCoveredApis` which will reside in each feature root folder and will mark the APIs that are not currently testable as `tested`.
3. We introduced the `checkNightlyTestsCoverage` gradle task which will be executed as `analysis` step in our Gitlab pipeline. This task will basically call the newly developed test coverage script with a specific test coverage threshold (for now in Android we are at 0.54 coverage for the `dd-sdk-android` module.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

**Review by 0xnm — APPROVED** [review]
lgtm! one question: maybe we can rename `NotCoveredApis` to `NotCoverableApis`? because the former one sounds like a to-do list to me.

File: buildSrc/src/main/kotlin/com/datadog/gradle/config/NightlyTestsCoveragConfig.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.gradle.api.tasks.Exec
+
+const val TEST_COVERAGE_THRESHOLD = 0.54
+fun Project.nightlyTestsCoverageConfig() {
+
+    tasks.create("checkNightlyTestsCoverage", Exec::class.java) {
+        val generateApiTasks = allprojects.mapNotNull { it.tasks.findByName("generateApiSurface") }
+        setDependsOn(generateApiTasks)
+        val threshold = TEST_COVERAGE_THRESHOLD
+        setWorkingDir(".")
```
[the remark this answers is not in the document]
Comment (mariusc83) [reply buildSrc/src/main/kotlin/com/datadog/gradle/config/NightlyTestsCoveragConfig.kt:19]: good point...I actually did not try with the default one. Will try it and let you know, eventually update the PR if the default one is the main directory which normally should be the case.

File: buildSrc/src/main/kotlin/com/datadog/gradle/config/NightlyTestsCoveragConfig.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.gradle.config
+
+import org.gradle.api.Project
+import org.gradle.api.tasks.Exec
+
+const val TEST_COVERAGE_THRESHOLD = 0.54
```
Comment (xgouchet) [inline buildSrc/src/main/kotlin/com/datadog/gradle/config/NightlyTestsCoveragConfig.kt:12]: This value should not be harcoded in the `buildSrc` module. Instead it should be in a `build.gradle/kts` file.
I would also define this task in a gradle script file too as it's only used in one place, but you can keep it here if it's more convenient

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/com/datadog/gradle/config/NightlyTestsCoveragConfig.kt (hunk 2 above)
Comment (mariusc83) [reply buildSrc/src/main/kotlin/com/datadog/gradle/config/NightlyTestsCoveragConfig.kt]: I'd rather keep it here for now. I made the threshold a variable of that function.

**Review by xgouchet — APPROVED** [review]
